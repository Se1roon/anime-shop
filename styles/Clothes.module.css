.section {
  margin-block: 10rem 3rem;
  display: flex;
  flex-direction: column;
  align-items: center;
}

.imgContainer {
  width: min(80%, 22rem);
  height: auto;
}

.imgContainer img {
  width: 100%;
  border-radius: 2rem;
  box-shadow: 0 0 0.8rem 2px var(--clr-accent);
}

.content {
  font-family: "Montserrat", sans-serif;
  color: var(--bg-light);
  margin-top: 1.2rem;
  display: flex;
  flex-direction: column;
  align-items: center;
}

.content .price {
  font-size: 1.1em;
}

.sizes {
  width: 100%;
  margin-top: 1rem;
  display: grid;
  place-content: center;
  grid-template-areas:
    "XS S"
    "M L"
    "XL XL";
  gap: 0.35rem;
}

.sizes > div {
  cursor: pointer;
  width: 4rem;
  aspect-ratio: 1;
  border: 2px solid var(--bg-light);
  border-radius: 1rem;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  transition: border-color 0.3s ease;
}

.sizes > div:hover {
  border-color: var(--clr-accent);
}

.sizes > .clicked {
  background-color: var(--clr-accent);
  border-color: var(--bg-dark);
}

.xs {
  grid-area: XS;
}

.s {
  grid-area: S;
}

.m {
  grid-area: M;
}

.l {
  grid-area: L;
}

.xl {
  justify-self: center;
  grid-area: XL;
}

.rating {
  margin-block: 0.4rem 1rem;
  display: flex;
  justify-content: center;
}

.rating > * {
  width: 1.5rem;
  aspect-ratio: 1;
}

.buy {
  cursor: pointer;
  font-family: "Montserrat", sans-serif;
  font-size: 1.2rem;
  color: var(--bg-light);
  background: transparent;
  margin-top: 1.5rem;
  padding: 0.4rem 1.1rem;
  outline: 0;
  border: 2px solid var(--clr-accent);
  border-radius: 1em;
  transition: all 0.3s ease;
}

.buy:hover,
.buy:focus {
  background-color: var(--clr-accent);
  box-shadow: 0 0 1rem 1px var(--clr-accent);
}

@media (min-width: 870px) {
  .section {
    margin-top: 15rem;
    flex-direction: row;
    justify-content: space-evenly;
  }
}

@media (min-width: 1100px) {
  .section {
    margin-inline: 10%;
  }
}

@media (min-width: 1300px) {
  .section {
    margin-inline: 20%;
  }
}
