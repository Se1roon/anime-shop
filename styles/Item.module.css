.section {
  margin-block: 10rem 3rem;
  display: flex;
  flex-direction: column;
  align-items: center;
}

@media (min-width: 870px) {
  .section {
    margin-top: 15rem;
    flex-direction: row;
    justify-content: space-evenly;
  }
}

@media (min-width: 1100px) {
  .section {
    margin-inline: 10%;
  }
}

@media (min-width: 1300px) {
  .section {
    margin-inline: 20%;
  }
}

.imgContainer {
  width: min(80%, 22rem);
  height: auto;
}

.img {
  width: 100%;
  border-radius: 2rem;
  box-shadow: 0 0 0.8rem 2px var(--clr-accent);
}

.content {
  font-family: "Montserrat", sans-serif;
  color: var(--bg-light);
  margin-top: 1.2rem;
  display: flex;
  flex-direction: column;
  align-items: center;
}

.price {
  font-size: 1.1em;
}

.rating {
  margin-block: 0.4rem 1rem;
  display: flex;
  justify-content: center;
}

.rating > * {
  width: 1.5rem;
  aspect-ratio: 1;
}

.sizes {
  width: 100%;
  margin-top: 1rem;
  display: grid;
  place-content: center;
  grid-template-areas:
    "XS S"
    "M L"
    "XL XL";
  gap: 0.35rem;
}

.sizes > div {
  cursor: pointer;
  width: 4rem;
  aspect-ratio: 1;
  border: 2px solid var(--bg-light);
  border-radius: 1rem;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  transition: border-color 0.3s ease;
}

.sizes > div:hover {
  border-color: var(--clr-accent);
}

.sizes > .clicked {
  background-color: var(--clr-accent);
  border-color: var(--bg-dark);
}

.xs {
  grid-area: XS;
}

.s {
  grid-area: S;
}

.m {
  grid-area: M;
}

.l {
  grid-area: L;
}

.xl {
  grid-area: XL;
  justify-self: center;
}

.btn {
  cursor: pointer;
  font-family: "Montserrat", sans-serif;
  font-size: 1.2rem;
  color: var(--bg-light);
  background: transparent;
  margin-top: 1.5rem;
  padding: 0.4rem 1.1rem;
  outline: 0;
  border: 2px solid var(--clr-accent);
  border-radius: 1em;
  transition: all 0.3s ease;
}

.btn:hover,
.btn:focus {
  background-color: var(--clr-accent);
  box-shadow: 0 0 1rem 1px var(--clr-accent);
}
