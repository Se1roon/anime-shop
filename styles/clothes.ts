import styled from "styled-components";

export const Section = styled.section`
  margin-block: 10rem 3rem;
  display: flex;
  flex-direction: column;
  align-items: center;

  @media (min-width: 870px) {
    margin-top: 15rem;
    flex-direction: row;
    justify-content: space-evenly;
  }

  @media (min-width: 1100px) {
    margin-inline: 10%;
  }

  @media (min-width: 1300px) {
    margin-inline: 20%;
  }
`;

export const ImgContainer = styled.div`
  width: min(80%, 22rem);
  height: auto;
`;

export const StyledImage = styled.img`
  width: 100%;
  border-radius: 2rem;
  box-shadow: 0 0 0.8rem 2px var(--clr-accent);
`;

export const Content = styled.div`
  font-family: "Montserrat", sans-serif;
  color: var(--bg-light);
  margin-top: 1.2rem;
  display: flex;
  flex-direction: column;
  align-items: center;
`;

export const Price = styled.p`
  font-size: 1.1em;
`;

export const Rating = styled.div`
  margin-block: 0.4rem 1rem;
  display: flex;
  justify-content: center;

  & > * {
  width: 1.5rem;
  aspect-ratio: 1;
`;

export const SizesContainer = styled.div`
  width: 100%;
  margin-top: 1rem;
  display: grid;
  place-content: center;
  grid-template-areas:
    "XS S"
    "M L"
    "XL XL";
  gap: 0.35rem;

  & > div {
    cursor: pointer;
    width: 4rem;
    aspect-ratio: 1;
    border: 2px solid var(--bg-light);
    border-radius: 1rem;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    transition: border-color 0.3s ease;
  }

  & > div:hover {
    border-color: var(--clr-accent);
  }

  & > .clicked {
    background-color: var(--clr-accent);
    border-color: var(--bg-dark);
  }
`;

export const XS = styled.div`
  grid-area: XS;
`;

export const S = styled.div`
  grid-area: S;
`;

export const M = styled.div`
  grid-area: M;
`;

export const L = styled.div`
  grid-area: L;
`;

export const XL = styled.div`
  justify-self: center;
  grid-area: XL;
`;

export const BuyBtn = styled.button`
  cursor: pointer;
  font-family: "Montserrat", sans-serif;
  font-size: 1.2rem;
  color: var(--bg-light);
  background: transparent;
  margin-top: 1.5rem;
  padding: 0.4rem 1.1rem;
  outline: 0;
  border: 2px solid var(--clr-accent);
  border-radius: 1em;
  transition: all 0.3s ease;

  &:hover,
  &:focus {
    background-color: var(--clr-accent);
    box-shadow: 0 0 1rem 1px var(--clr-accent);
  }
`;
