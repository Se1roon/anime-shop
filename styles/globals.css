@import url("https://fonts.googleapis.com/css2?family=Montserrat:wght@500&display=swap");
@import url("https://fonts.googleapis.com/css2?family=Poppins:wght@300;500;600&display=swap");

:root {
  --bg-light: hsl(0 0% 100%);
  --bg-dark: hsl(0 0% 10%);
  --clr-accent: hsl(300 100% 50%);
}

*,
*::after,
*::before {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
}

::-webkit-scrollbar {
  width: 0.3rem;
  height: 0.3rem;
}

::-webkit-scrollbar-thumb {
  background-color: var(--clr-accent);
  border-radius: 100%;
}

body {
  background-color: var(--bg-dark);
  height: 100vh;
}

a {
  color: inherit;
  text-decoration: none;
}

* {
  box-sizing: border-box;
}

.swiper {
  width: 950px;
}

.swiper-pagination-bullet {
  height: 7px;
  width: 26px;
  border-radius: 25px;
  background-color: #7d2ae8;
}

.swiper-button-prev,
.swiper-button-next {
  opacity: 0.6;
  transition: all 0.3s ease;
}

.swiper-button-prev::after,
.swiper-button-next::after {
  color: var(--bg-dark);
}

.swiper-button-prev:hover,
.swiper-button-next:hover {
  opacity: 1;
}

@media (min-width: 1100px) {
  body {
    background-color: var(--bg-light);
  }
}
