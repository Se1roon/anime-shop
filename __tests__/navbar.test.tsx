import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import Navbar from "../components/navbar";
import styles from "../components/css/Navbar.module.css";

describe("Navbar component", () => {
  it("checks whether everything renders correctly", () => {
    const { container } = render(<Navbar />);

    expect(container).toMatchSnapshot();
  });

  it("check whether .visible is added after menu click", async () => {
    render(<Navbar />);
    const menu = document.querySelector(`.${styles.menu}`);

    await userEvent.click(menu);

    const nav = document.querySelector(`.${styles.nav}`);
    expect(nav.classList.contains(`${styles.visible}`)).toBe(true);
  });
});
