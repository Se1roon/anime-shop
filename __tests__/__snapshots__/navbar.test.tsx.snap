// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component checks whether everything renders correctly 1`] = `
<div>
  <header
    class="header"
  >
    <div
      class="container"
    >
      <img
        alt="Logo"
        class="logo"
        data-nimg="1"
        decoding="async"
        height="64"
        loading="lazy"
        src="/_next/image?url=%2F..%2Fpublic%2Fimages%2Flogo.png&w=128&q=75"
        srcset="/_next/image?url=%2F..%2Fpublic%2Fimages%2Flogo.png&w=64&q=75 1x, /_next/image?url=%2F..%2Fpublic%2Fimages%2Flogo.png&w=128&q=75 2x"
        style="color: transparent;"
        width="64"
      />
      <div
        class="menu"
      >
        <div />
        <div />
        <div />
      </div>
    </div>
    <nav
      class="nav"
    >
      <ul
        class="links"
      >
        <li>
          <a
            href="/hoodies"
          >
            Hoodies
          </a>
        </li>
        <li>
          <a
            href="/shirts"
          >
            Shirts
          </a>
        </li>
        <li>
          <a
            href="/pants"
          >
            Pants
          </a>
        </li>
      </ul>
      <a
        class="cta"
        href="/contact"
      >
        <button
          class="btn"
        >
          Contact
        </button>
      </a>
    </nav>
  </header>
</div>
`;
